Jenkinsfile

```diff
@@ -1,13 +1,13 @@
-def comonVersion = "2.356"
-def recentLTS = "2.346.1"
+def comonVersion = "2.362"
```

Comment:
If I understand correctly the original goal of this CI config was to have a testing matrix that would check the plugin with the min and the max version of Jenkins that the plugin supports.
If this is still the intention, then we should change this version to `2.453` (the latest release at the moment, still seems to support Java 11).

While we're at it, we could also fix the "comon" typo (and change it to "common" or even "latest").

Also, the original matrix tested the plugin with two different Java versions: 8 and 11. I think it makes sense to continue doing so. We can change the JDK of the "common version" builds to 17.

Reply:
Changing the latestSupported version to 2.453 requires manual version changes below:
<img width="1018" alt="Screenshot 2024-04-12 at 10 02 10 AM" src="https://github.com/jenkinsci/datadog-plugin/assets/144174402/7e4331d2-aaa9-4da8-8211-ba98f743835e">

However, including these version bumps causes the build to break for the recentLTS version--as such I decided to only bump up the latestSupported to the max version that could be built without breaking the recentLTS version.